plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/navigator/DiagramCreateWizard.java

```diff
@@ -90,13 +94,16 @@ public void init(IWorkbench workbench, IStructuredSelection selection) {
                 }
             }
         }
-        this.folder = diagramFolder;
+        if (diagramFolder != null) {
+            this.folder = diagramFolder;
+            this.project = diagramFolder == null ? null : DBPPlatformDesktop.getInstance().getWorkspace().getProject(diagramFolder.getProject());
```

Comment:
`diagramFolder` can't be null here

Reply:
It can be null, because `DesktopWorkspaceImpl:getResourceDefaultRoot` may return null. Why don't you think so?